src/fontcache.cpp

```diff
@@ -1056,9 +1056,300 @@ static void LoadWin32Font(FontSize fs)
 	new Win32FontCache(fs, logfont, settings->size);
 }
 
+#elif defined(WITH_COCOA)
```

Comment:
Think this file needs splitting into per-os variants. It's getting a bit silly

Reply:
Yes, except of course that e.g. FreeType is actually not a real OS-variant. Both OSX and Win will still use FreeType if built with it.

If you have a good idea how to organise the split (probably including fontdetection.cpp), I'm willing to do it.

Reply:
I suppose it would be per rendering engine, rather than per-OS, a bit like we have SDL, Windows, Mac video drivers.

I would be quite keen to get native font rendering in on Mac if we can, to avoid the freetype dependency. That said, I haven't yet tested this PR, but hope to do so when I get a chance.